cpp/src/Ice/Service.cpp

```diff
@@ -1496,25 +1497,25 @@ ServiceStatusManager::changeStatus(DWORD state, DWORD controlsAccepted)
 void
 ServiceStatusManager::reportStatus()
 {
-    Lock sync(*this);
+    lock_guard lock(_mutex);
 
     SetServiceStatus(_handle, &_status);
 }
 
 void
 ServiceStatusManager::run()
 {
-    Lock sync(*this);
+    unique_lock lock(_mutex);
 
-    IceUtil::Time delay = IceUtil::Time::milliSeconds(1000);
+    auto delay = chrono::milliseconds(1000);
     _status.dwWaitHint = 2000;
     _status.dwCheckPoint = 0;
 
     while(!_stopped)
     {
         _status.dwCheckPoint++;
         SetServiceStatus(_handle, &_status);
-        timedWait(delay);
+        _conditionVariable.wait_until(chrono::system_clock::now() + delay);
```

Comment:
I don't understand why you use wait_until instead of wait_for in this code (and other translations of timedWait).

Then, if we keep wait_for, this should be steady_lock, not system_clock.

Reply:
I will replace it, I missed there is a `wait_for` method.